apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4SamhsaMatcher.java

```diff
@@ -522,43 +523,89 @@ private static String normalizeIcd10Code(String icd10DiagnosisCode) {
+   *     {@link Coding} with the HCPCS {@link System}, <code>false</code> if it does not
+   */
+  private boolean findHcpcsCoding(CodeableConcept procedureConcept) {
     for (Coding procedureCoding : procedureConcept.getCoding()) {
       if (TransformerConstants.CODING_SYSTEM_HCPCS.equals(procedureCoding.getSystem())) {
-        if (isSamhsaCptCode(procedureCoding)) return true;
-      } else {
-        /*
-         * Fail safe: if we don't know the procedure Coding system, assume the code is
-         * SAMHSA.
-         */
         return true;
       }
     }
-
-    // No blacklisted procedure Codings found: this procedure isn't SAMHSA-related.
     return false;
   }
 
   /**
-   * @param procedureCoding the procedure {@link Coding} to check
-   * @return <code>true</code> if the specified procedure {@link Coding} matches one of the {@link
-   *     #cptCodes} entries, <code>false</code> if it does not
+   * @param procedureConcept the procedure {@link CodeableConcept} to check
+   * @return <code>false</code> if the specified procedure {@link CodeableConcept} contains any
+   *     {@link Coding}s that do not contain any of the {@link System}s <code>true</code> if they do
    */
-  private boolean isSamhsaCptCode(Coding procedureCoding) {
+  private boolean hasUnknownSystem(CodeableConcept procedureConcept) {
+    Set<String> codingSystems =
+        procedureConcept.getCoding().stream()
+            .map(coding -> coding.getSystem())
+            .collect(Collectors.toSet());
+    // CHeck for HCPCS and HCPCS CD code systems, so that the right code is given in a backwards
+    // compatible way: https://jira.cms.gov/browse/BFD-1345
+    if (codingSystems.size() == 2
+        && codingSystems.contains(TransformerConstants.CODING_SYSTEM_HCPCS)
+        && codingSystems.contains(
+            TransformerUtilsV2.calculateVariableReferenceUrl(CcwCodebookVariable.HCPCS_CD))) {
+      return false;
+    } else if (codingSystems.size() == 1
+        && codingSystems.contains(TransformerConstants.CODING_SYSTEM_HCPCS)) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  /**
+   * @param procedureConcept the procedure {@link CodeableConcept} to check
+   * @return <code>true</code> if the specified procedure {@link CodeableConcept} contains any
+   *     {@link Coding}s that match any of the {@link #cptCodes}, <code>false</code> if they all do
+   *     not
+   */
+  private boolean isSamhsaCptCode(CodeableConcept procedureConcept) {
     /*
      * Note: CPT codes represent a subset of possible HCPCS codes (but are the only
      * subset that we blacklist from).
      */
-    if (!TransformerConstants.CODING_SYSTEM_HCPCS.equals(procedureCoding.getSystem()))
-      throw new IllegalArgumentException();
+    for (Coding procedureCoding : procedureConcept.getCoding()) {
+      if (!TransformerConstants.CODING_SYSTEM_HCPCS.equals(procedureCoding.getSystem())) {
+        throw new IllegalArgumentException();
+      }
```

Comment:
This causes an issue if any of the system values are not HPCPCS (which they will always be, because we're adding both HCPCS and HCPCS_CS). The check here should allow for both systems codes, since we are adding both.